localstack/services/cloudformation/models/iam.py

```diff
@@ -221,16 +348,110 @@ def get_physical_resource_id(self, attribute=None, **kwargs):
 
     @staticmethod
     def get_deploy_templates():
+        def _add_roles(resource_id, resources, resource_type, func, stack_name):
+            client = aws_stack.connect_to_service("iam")
+            resource = resources[resource_id]
+            props = resource["Properties"]
+            if props.get("Roles", []):
+                client.add_role_to_instance_profile(
+                    InstanceProfileName=props["InstanceProfileName"],
+                    RoleName=props["Roles"][0],
```

Comment:
The [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-instanceprofile.html#cfn-iam-instanceprofile-roles) are not super clear on this:

> The name of the role to associate with the instance profile. Only one role can be assigned to an EC2 instance at a time, and all applications on the instance share the same role and permissions. 

... but should we loop over `props["Roles"]`, or do we always only add the first one to the instance profile?

Reply:
Just tested this on AWS and it enforces a list length of 1.

```
7:29:07 AM | CREATE_FAILED        | AWS::IAM::InstanceProfile | instanceprofile
Roles has too many elements. The limit is 1.
```

So we might also just throw an error here if the list length is > 1